discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/KubernetesApiServiceDiscovery.scala

```diff
@@ -30,18 +31,26 @@ import akka.event.Logging
 object KubernetesApiServiceDiscovery {
 
   /**
+   * INTERNAL API
+   *
    * Finds relevant targets given a pod list. Note that this doesn't filter by name as it is the job of the selector
    * to do that.
    */
+  @InternalApi
   private[kubernetes] def targets(podList: PodList,
-                                  portName: String,
+                                  portName: Option[String],
                                   podNamespace: String,
                                   podDomain: String): Seq[ResolvedTarget] =
     for {
       item <- podList.items
       if item.metadata.flatMap(_.deletionTimestamp).isEmpty
       container <- item.spec.toVector.flatMap(_.containers)
-      port <- container.ports.getOrElse(Seq.empty).find(_.name.contains(portName))
+      port <- portName match {
```

Comment:
So if using service discovery not with bootstrap and not passing in a port this will result in all the ports being returned?

Reply:
Yes. Indeed this only really makes sense when the service only exposes one port, or when the results are post-processed like in #503 .